nomad/server.go

```diff
@@ -1399,6 +1409,42 @@ func (s *Server) setupRaft() error {
 	return nil
 }
 
+// checkRaftVersionFile reads the Raft version file and returns an error if
+// the Raft version is incompatible with the current version configured.
+// Provide best-effort check if the file cannot be read.
+func (s *Server) checkRaftVersionFile(path string) error {
+	raftVersion := s.config.RaftConfig.ProtocolVersion
+	baseWarning := "use the 'nomad operator raft list-peers' command to make sure the Raft protocol versions are consistent"
+
+	_, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			s.logger.Info(fmt.Sprintf("Raft version file not found, %v", baseWarning))
+		} else {
+			s.logger.Warn(fmt.Sprintf("unable to read Raft version file, %s", baseWarning), "error", err)
+		}
+		return nil
+	}
+
+	v, err := ioutil.ReadFile(path)
+	if err != nil {
+		s.logger.Warn(fmt.Sprintf("unable to read Raft version file, %s", baseWarning), "error", err)
```

Comment:
In this case and the case where we can't stat the file but it exists, is the `baseWarning` the right action to recommend? It seems like we want them to remove the old raft version file too?

Reply:
Good question, I'm not sure to be honest. Originally I had it so any error here would bubble up and prevent the server from starting, but that seemed not great.

In the case that we can't read the version file, I didn't return an error to allow the file to be overwritten. If the problem is some kind of hardware failure, that operation will result in an error. If the problem is the file content, we just overwrite it with the new value.

So the goal of `baseWarning` is alert users that we weren't able to verify the previous Raft version, so they should do it themselves. Maybe the `baseWarning` message itself should be more explicit?

Reply:
Ah yeah I see what you mean about how narrowly you scoped when to return an error. So on second thought, I think what you've got here LGTM